src/Agent.Plugins/PipelineCache/PipelineCacheServer.cs

```diff
@@ -156,12 +159,21 @@ public class PipelineCacheServer
             string targetDirectory,
             CancellationToken cancellationToken)
         {
+
             DownloadDedupManifestArtifactOptions options = DownloadDedupManifestArtifactOptions.CreateWithManifestId(
                 manifestId,
                 targetDirectory,
                 proxyUri: null,
                 minimatchPatterns: null);
-            return dedupManifestClient.DownloadAsync(options, cancellationToken);
+            using( Process tarring = new Process())
```

Comment:
Need to gracefully handle scenarios where `tar` is not found, can't be executed for some reason, or just fails.

Reply:
Done.